test/integration/filters/local_reply_during_encoding_filter.cc

```diff
@@ -0,0 +1,30 @@
+#include <string>
+
+#include "envoy/http/filter.h"
+#include "envoy/registry/registry.h"
+#include "envoy/server/filter_config.h"
+
+#include "extensions/filters/http/common/pass_through_filter.h"
+
+#include "test/extensions/filters/http/common/empty_http_filter_config.h"
+#include "test/integration/filters/common.h"
+
+namespace Envoy {
+
+class LocalReplyDuringEncode : public Http::PassThroughFilter {
+public:
+  constexpr static char name[] = "local-reply-during-encode";
+
+  Http::FilterHeadersStatus encodeHeaders(Http::ResponseHeaderMap&, bool) override {
+    decoder_callbacks_->sendLocalReply(Http::Code::InternalServerError, "", nullptr, absl::nullopt,
```

Comment:
Would it make sense to add `sendLocalReply` to the `StreamEncoderFilterCallbacks` interface?

Reply:
+1 per slack convo also